src/systems/physics/Physics.cc

```diff
@@ -1312,6 +1344,34 @@ void PhysicsPrivate::CreateCollisionEntities(const EntityComponentManager &_ecm,
         }
         auto linkPtrPhys = this->entityLinkMap.Get(_parent->Data());
 
+        if (const auto existingShape = linkPtrPhys->GetShape(_name->Data()))
+        {
+          // No need to create this collision shape because it was already
+          // created when parsing the model.
+
+          // TODO(MXG): There's some significant design inconsistency in this
+          // CreateCollisionEntities function. There's a mix of using the
+          // ConstructSdfCollision feature and other collision attachment
+          // features like AttachHeightMapFeature and AttachMeshShape. This
+          // makes it confusing for physics plugin implementers to know where
+          // and how to implement the construction of collision shapes.
+          //
+          // In the current implementation, if a physics engine implements
+          // constructing a mesh or heightmap in ConstructSdfCollision then that
+          // ability won't actually get leveraged by gz-sim because it
+          // specifically uses AttachMeshShapeFeature and AttachHeightMapFeature
+          // for those shapes. The physics plugin implementer needs to know to
+          // implement those specific features instead.
+          auto linkCollisionFeature =
+              this->entityLinkMap.EntityCast<CollisionFeatureList>(
+                  _parent->Data());
+          this->entityCollisionMap.AddEntity(
+            _entity, linkCollisionFeature->GetCollision(_name->Data()));
```

Comment:
I changed `GetCollision` to `GetShape` on this line, and it works fine for me. I think we can revert the `GetCollision` API from `gz/physics/sdf/ConstructCollision.hh`

Reply:
https://github.com/gazebosim/gz-physics/pull/444